apps/explorer/src/pages/validator/ValidatorDetails.tsx

```diff
@@ -0,0 +1,259 @@
+
+import ErrorResult from '~/components/error-result/ErrorResult';
+import { StatsCoin } from '~/components/top-validators-card/StatsCoin';
+import { useGetObject } from '~/hooks/useGetObject';
+import {
+    VALIDATORS_OBJECT_ID,
+    type ValidatorState,
+} from '~/pages/validator/ValidatorDataTypes';
+import { Card } from '~/ui/Card';
+import { DescriptionList, DescriptionItem } from '~/ui/DescriptionList';
+import { Heading } from '~/ui/Heading';
+import { ImageIcon } from '~/ui/ImageIcon';
+import { AddressLink } from '~/ui/InternalLink';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Stats } from '~/ui/Stats';
+import { Text } from '~/ui/Text';
+import { getName } from '~/utils/getName';
+import { getStakedPercent } from '~/utils/getStakedPercent';
+
+function ValidatorDetails() {
+    const { id } = useParams();
+    const { data, isLoading } = useGetObject(VALIDATORS_OBJECT_ID);
+
+    const validatorsData =
+        data &&
+        is(data.details, SuiObject) &&
+        data.details.data.dataType === 'moveObject'
+            ? (data.details.data.fields as ValidatorState)
+            : null;
+
+    const validatorData = useMemo(() => {
+        if (!validatorsData) return null;
+        return (
+            validatorsData.validators.fields.active_validators.find(
+                (av) => av.fields.metadata.fields.sui_address === id
+            ) || null
+        );
+    }, [id, validatorsData]);
+
+    const validator = useMemo(() => {
+        if (!validatorData || !validatorsData) return null;
+
+        const { name, pubkey_bytes, sui_address } =
+            validatorData.fields.metadata.fields;
+
+        const { sui_balance, starting_epoch, delegation_token_supply } =
+            validatorData.fields.delegation_staking_pool.fields;
+
+        const num_epochs_participated = validatorsData.epoch - starting_epoch;
+
+        const APY = Math.pow(
+            1 +
+                (sui_balance - delegation_token_supply.fields.value) /
+                    delegation_token_supply.fields.value,
+            365 / num_epochs_participated - 1
+        );
+
+        return {
+            name: getName(name),
+            pubkeyBytes: new Base64DataBuffer(
+                new Uint8Array(pubkey_bytes)
+            ).toString(),
+            suiAddress: sui_address,
+            apy: APY ? APY : 0,
+            logo: null,
+            delegatedStakePercentage: getStakedPercent(
+                validatorData.fields.stake_amount,
+                validatorsData.validators.fields.total_validator_stake
+            ),
+            totalStake: validatorData.fields.stake_amount,
+            delegatedStake: validatorData.fields.stake_amount,
+            address: sui_address,
+
+            // TODO: add missing fields
+            numberOfDelegators: 0,
+            selfStake: 0,
+            lastEpoch: 0,
+            totalRewards: 0,
+        };
+    }, [validatorData, validatorsData]);
+
+    if (!id) {
```

Comment:
Is this actually possible? Seems like it's not and we can just cast the type if we need it narrowed.